core/types/receipt.go

```diff
@@ -54,25 +56,58 @@ func NewReceipt(root []byte, cumulativeGasUsed *big.Int) *Receipt {
 	return &Receipt{PostState: common.CopyBytes(root), CumulativeGasUsed: new(big.Int).Set(cumulativeGasUsed), Status: TxStatusUnknown}
 }
 
+type receiptRLP struct {
+	PostStateOrStatus []byte
+	CumulativeGasUsed *big.Int
+	Bloom             Bloom
+	Logs              vm.Logs
+}
+
 // EncodeRLP implements rlp.Encoder, and flattens the consensus fields of a receipt
 // into an RLP stream.
 func (r *Receipt) EncodeRLP(w io.Writer) error {
-	return rlp.Encode(w, []interface{}{r.PostState, r.CumulativeGasUsed, r.Bloom, r.Logs})
+	var stateOrStatus []byte
+	if len(r.PostState) == 0 { // if there is no state, EIP-658 is in place
+		if r.Status != TxStatusUnknown {
+			stateOrStatus = []byte{byte(r.Status)}
+		} else {
+			return errors.New("invalid receipt: PostState not present (EIP-658?) but transaction Status is unknown")
```

Comment:
I think this is accurate to the spec.

Since we're essentially unable to send transactions that our client hasn't "run," is this potentially a bottleneck if we wanted the client to be able to "forward" transactions that it hadn't run (maybe tx pool is functionally separated from block validation). Not a big thing, just noting that encoding only allowed for "GOOD/BAD" instead of "GOOD/BAD/IDK".

Reply:
IMO EIP-658 doesn't allow "IDK" ;)
And without EIP-658 `PostState` has to be set.

Reply:
Yea, agreed. Just noting.